### PR TITLE
distributions.Categorical support logits not None

### PR DESCRIPTION
**Pull request by Ldpe2G**

### CODE REVIEW

**Review by doombeaker — APPROVED** [review]
简单测了下，用了几个case试，已经和 torch 对齐了。
先approve了。
不过 whisper 模型的行为还不对齐（不一定和这个算子有关），正在二分定位哪里开始不同，后续如果有相关问题我再继续反馈

**Review by Flowingsun007 — APPROVED** [review]
